src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+  tooltip: {
+    background: 'rgba(0, 0, 0, .8)',
+    color: 'white',
+    borderRadius: '4px',
+    pointerEvents: 'none',
+  },
+  tooltipKey: {
+    display: 'inline-block',
+    width: '10px',
+    height: '10px',
+    marginRight: '10px',
+  },
+};
+
+function Footer({ dataset, index, currData }) {
```

Comment:
In-lining is better:

```javascript
import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  tooltip: {
    background: 'rgba(0, 0, 0, .8)',
    color: 'white',
    borderRadius: '4px',
    pointerEvents: 'none',
  },
  tooltipKey: {
    display: 'inline-block',
    width: '10px',
    height: '10px',
    marginRight: '10px',
  },
};

function CustomTooltip({ classes, tooltipModel, series }) {
  if (tooltipModel.opacity === 0) return null;

  const firstPoint = tooltipModel.dataPoints[0];
  const labelColors = tooltipModel.labelColors[0];
  const { xLabel, index, datasetIndex } = firstPoint;
  const seriesLabel = series[datasetIndex].label;
  const higherIsBetter = !series[datasetIndex].meta.lower_is_better;
  const higherOrLower = higherIsBetter ? 'higher is better' : 'lower is better';
  const paddingStyle = {
    padding: `${tooltipModel.yPadding}px ${tooltipModel.xPadding}px`,
  };
  let footer = null;

  if (index > 0) {
    const [prev, curr] = series[datasetIndex].data.slice(index - 1);
    const delta = curr.value - prev.value;
    const deltaPercentage = (delta / prev.value) * 100;
    const hgURL = `https://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=${
      prev.revision
    }&tochange=${curr.revision}`;

    footer = (
      <React.Fragment>
        <div>
          Δ {delta.toFixed(2)} ({deltaPercentage.toFixed(0)} %)
        </div>
        <div>
          <a href={hgURL}>{curr.revision.slice(0, 12)}</a>
        </div>
      </React.Fragment>
    );
  }

  return (
    <div className={classes.tooltip} style={paddingStyle}>
      <div>{xLabel}</div>
      <div>
        <span style={labelColors} className={classes.tooltipKey} />
        {seriesLabel}: {firstPoint.yLabel}
      </div>
      <div>
        {firstPoint.yLabel} ({higherOrLower})
      </div>
      {footer}
    </div>
  );
}

export default withStyles(styles)(CustomTooltip);

```

Reply:
Got it, just some things to point out:
- `tooltipModel.dataPoints[0]`doesn't refer to the first data point in the series, as you can check `tooltip`in [this gist](https://gist.github.com/2alin/2a1ef853d3dc9d251bb398b1f624c422).`tooltipModel.dataPoints` is an array with always one element that has the information of the current point, so I think it would be better to define it as `const currPoint = tooltipModels.dataPoints[0]`
- For `deltaPercentage`, I'm not sure that we should round it to an integer (`deltaPercentage.toFixed(0)`), there are many points in the chart (just checked it already) that gives a deltaPercentage of -0.38, 0.38, etc. I will round it to one decimal position, as it's set in the [treeherder graphs](https://treeherder.mozilla.org/perf.html#/graphs?series=mozilla-central,1775979,1,10&series=mozilla-central,1776166,1,10&series=mozilla-central,1776360,1,10&selected=mozilla-central,1775979,438078,733182131,10), but if you insist I'll set it as suggested then.
- 

About this:
> delay any formatting (eg .toFixed(2)) to the last possible line

I had it like that in a first version of another PR, but I was suggested to move it to how it was now. I suppose your reason is to keep the calculated value as precise as possible, and only trim it when it's displayed.

Reply:
* On the subject of `const currPoint = tooltipModels.dataPoints[0]`: Feel free to pick better names.
* For `deltaPercentage`, there are already 2 significant digits to the left of the decimal point.  Parts of a percent is unreasonably precise.  But matching Treeherder is fine too.
* On ` .toFixed(2)`, it is translating to a humane, but inaccurate, format. You will notice I can use the `delta` to calculate the percentage because I did not loose accuracy. We also keep the translation code together; in this case all in the JSX.